front/src/modules/simulationResult/components/SpeedSpaceChart/helpers.ts

```diff
@@ -0,0 +1,155 @@
+/* eslint-disable import/prefer-default-export */
+import type {
+  LayerData,
+  Data,
+  ElectrificationValues,
+  PowerRestrictionValues,
+  ElectricalPofilelValues,
+  SpeedLimitTagValues,
+} from '@osrd-project/ui-speedspacechart/dist/types/chartTypes';
+
+import type {
+  PathPropertiesFormatted,
+  PositionData,
+  SimulationResponseSuccess,
+} from 'applications/operationalStudies/types';
+import type { ReportTrainV2 } from 'common/api/generatedEditoastApi';
+import { mmToKm, msToKmh, mToKm } from 'utils/physics';
+
+import { electricalProfilesDesignValues } from './consts';
+
+export const formatSpeeds = (simulation: ReportTrainV2) => {
+  const { positions, speeds } = simulation;
+  return speeds.map((value, index) => ({
+    position: {
+      start: mmToKm(positions[index]),
+    },
+    value: msToKmh(value),
+  }));
+};
+
+export const formatStops = (operationalPoints: PathPropertiesFormatted['operationalPoints']) =>
+  operationalPoints.map(({ position, extensions: { identifier, sncf } = {} }) => ({
+    position: {
+      start: mmToKm(position),
+    },
+    value:
+      identifier && sncf ? `${identifier.name} ${sncf.ch !== ('' || '00') ? sncf.ch : ''}` : '',
+  }));
+
+export const formatElectrifications = (
+  electrifications: PathPropertiesFormatted['electrifications']
+) =>
+  electrifications.map(({ electrificationUsage, start, stop }) => ({
+    position: {
+      start: mToKm(start),
+      end: mToKm(stop),
+    },
+    value: {
+      type: electrificationUsage.type,
+      voltage:
+        'voltage' in electrificationUsage
+          ? (electrificationUsage.voltage as ElectrificationValues['voltage'])
+          : undefined,
+      lowerPantograph:
+        'lower_pantograph' in electrificationUsage ? electrificationUsage.lower_pantograph : false,
+    },
+  }));
+
+export const formatSlopes = (slopes: PositionData<'gradient'>[]) => {
+  let actualSlopes: LayerData<number>[] = [];
+  if (slopes)
+    actualSlopes = slopes.map((slope, index) => ({
+      position: {
+        start: mToKm(index === 0 ? 0 : slopes[index - 1].position),
+        end: mToKm(slope.position),
+      },
+      value: slope.gradient,
+    }));
+  return actualSlopes;
+};
+
+export const getProfileValue = (
+  electricalProfileType: 'profile' | 'no_profile',
+  profile: keyof typeof electricalProfilesDesignValues | null,
+  voltage?: '1500V' | '25000V',
+  handled?: boolean
+) => {
+  if (electricalProfileType === 'no_profile') {
+    return { electricalProfile: 'neutral' };
+  }
+  if (!handled || profile === null) {
+    return {
+      electricalProfile: 'incompatible',
+      color:
+        voltage === '1500V'
+          ? electricalProfilesDesignValues.F.color
+          : electricalProfilesDesignValues['20000V'].color,
+    };
+  }
+  return {
+    electricalProfile: profile,
+    ...electricalProfilesDesignValues[profile!],
+  };
+};
```

Comment:
To use less parameters in this function, we could refacto it like that (and this will require to adapt tests also)

```suggestion
export const getProfileValue = (
  electricalProfileValue: SimulationResponseSuccess['electrical_profiles']['values'][number],
  voltage?: '1500V' | '25000V'
) => {
  if (electricalProfileValue.electrical_profile_type === 'no_profile') {
    return { electricalProfile: 'neutral' };
  }
  if (
    !electricalProfileValue.handled ||
    !electricalProfileValue.profile ||
    electricalProfileValue.profile === null
  ) {
    return {
      electricalProfile: 'incompatible',
      color:
        voltage === '1500V'
          ? electricalProfilesDesignValues.F.color
          : electricalProfilesDesignValues['20000V'].color,
    };
  }
  return {
    electricalProfile: electricalProfileValue.profile,
    ...electricalProfilesDesignValues[
      electricalProfileValue.profile as keyof typeof electricalProfilesDesignValues
    ],
  };
};
```

Reply:
done